### PR TITLE
Bumping version for 0.11.1

### PR DESCRIPTION
**Pull request by rachitchauhan43**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Type of changes**
Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**Feature/Issue validation/testing**:

Please describe the tests that you ran to verify your changes and relevant result summary. Provide instructions so it can be reproduced.
Please also list any relevant details for your test configuration.

- [ ] Test A
- [ ] Test B

- Logs

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Checklist**:

- [ ] Have you added unit/e2e tests that prove your fix is effective or that this feature works?
- [ ] Has code been commented, particularly in hard-to-understand areas?
- [ ] Have you made corresponding changes to the documentation?

**Release note**:

```release-note

```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -136,7 +136,7 @@ undeploy-dev: kustomize
 	${KUSTOMIZE} build config/overlays/development | kubectl delete -f -
 
 # Generate manifests e.g. CRD, RBAC etc.
-manifests: controller-gen kustomize
+manifests: controller-gen
```
Comment (ckadner) [inline Makefile]: Removing the `kustomize` dependency here seems to cause the Go build error `kustomize: not found`

```yaml
/bin/sh: 1: /home/runner/work/kserve/kserve/bin/kustomize: not found
make: *** [Makefile:169: manifests] Error 127
Error: Process completed with exit code 2.
```
https://github.com/kserve/kserve/actions/runs/6278191515/job/17051425230?pr=3141#step:6:330

File: python/VERSION [hunk 2]
```diff
@@ -1 +1 @@
-0.11.0
+0.11.1
```
Comment (ckadner) [inline python/VERSION]: should keep a newline at the end of file?

**Review by ckadner — CHANGES_REQUESTED** [review]
Thanks Rachit. Looks good overall. Just a few small nit picks / questions.

File: Makefile (hunk 1 above)
Comment (rachitchauhan43) [reply Makefile]: Oh, yeah. Sorry about that. It was my local change that wasn't intended to be pushed.

File: python/VERSION (hunk 2 above)
Comment (rachitchauhan43) [reply python/VERSION]: Thanks @yuzisun  for fixing these.

**Review by ckadner — APPROVED** [review]
Thanks Dan, Rachit. Looks good.

/lgtm

**Pull request merged**